Core/Node-API/Source/js_native_api_v8.cc

```diff
@@ -2208,59 +2498,130 @@ GEN_COERCE_FUNCTION(STRING, String, string)
-                                           finalize_cb,
-                                           finalize_hint,
-                                           result);
-}
-
-napi_status napi_unwrap(napi_env env, napi_value obj, void** result) {
+napi_status NAPI_CDECL napi_wrap(napi_env env,
+                                 napi_value js_object,
+                                 void* native_object,
+                                 napi_finalize finalize_cb,
+                                 void* finalize_hint,
+                                 napi_ref* result) {
+  return v8impl::Wrap(
+      env, js_object, native_object, finalize_cb, finalize_hint, result);
+}
+
+napi_status NAPI_CDECL napi_unwrap(napi_env env,
+                                   napi_value obj,
+                                   void** result) {
   return v8impl::Unwrap(env, obj, result, v8impl::KeepWrap);
 }
 
-napi_status napi_remove_wrap(napi_env env, napi_value obj, void** result) {
+napi_status NAPI_CDECL napi_remove_wrap(napi_env env,
+                                        napi_value obj,
+                                        void** result) {
   return v8impl::Unwrap(env, obj, result, v8impl::RemoveWrap);
 }
 
-napi_status napi_create_external(napi_env env,
-                                 void* data,
-                                 napi_finalize finalize_cb,
-                                 void* finalize_hint,
-                                 napi_value* result) {
+napi_status NAPI_CDECL napi_create_external(napi_env env,
+                                            void* data,
+                                            napi_finalize finalize_cb,
+                                            void* finalize_hint,
+                                            napi_value* result) {
   NAPI_PREAMBLE(env);
   CHECK_ARG(env, result);
 
   v8::Isolate* isolate = env->isolate;
 
   v8::Local<v8::Value> external_value = v8::External::New(isolate, data);
 
-  // The Reference object will delete itself after invoking the finalizer
-  // callback.
-  v8impl::Reference::New(env,
-      external_value,
-      0,
-      true,
-      finalize_cb,
-      data,
-      finalize_hint);
+  if (finalize_cb) {
+    // The Reference object will delete itself after invoking the finalizer
+    // callback.
+    v8impl::Reference::New(env,
+                           external_value,
+                           0,
+                           v8impl::Ownership::kRuntime,
+                           finalize_cb,
+                           data,
+                           finalize_hint);
+  }
 
   *result = v8impl::JsValueFromV8LocalValue(external_value);
 
   return napi_clear_last_error(env);
 }
 
-napi_status napi_get_value_external(napi_env env,
-                                    napi_value value,
-                                    void** result) {
-  CHECK_ENV(env);
+// Need NAPI_VERSION >= 8
+/*napi_status NAPI_CDECL napi_type_tag_object(napi_env env,
```

Comment:
Did we comment this out or is this part of the node code?

Reply:
we did because of missing napi_type_tag.

Reply:
It looks like the NAPI_VERSION >= 8 check was absent from this file. The functions were commented out in previous commits. I've changed it to a #if to be consistent with function declaration.